pkg/cache/cache_test.go

```diff
@@ -22,40 +22,90 @@ import (
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 	kcorev1 "k8s.io/api/core/v1"
+	kmetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	kcache "k8s.io/client-go/tools/cache"
 
 	"sigs.k8s.io/controller-runtime/pkg/cache"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+var informerCache cache.Cache
+var stop chan struct{}
+var knownPod1 runtime.Object
+var knownPod2 runtime.Object
+var knownPod3 runtime.Object
+
+func createPod(name, namespace string, restartPolicy kcorev1.RestartPolicy) runtime.Object {
+	three := int64(3)
+	pod := &kcorev1.Pod{
+		ObjectMeta: kmetav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"test-label": name,
+			},
+		},
+		Spec: kcorev1.PodSpec{
+			Containers:            []kcorev1.Container{{Name: "nginx", Image: "nginx"}},
+			RestartPolicy:         restartPolicy,
+			ActiveDeadlineSeconds: &three,
+		},
+	}
+	cl, err := client.New(cfg, client.Options{})
+	Expect(err).NotTo(HaveOccurred())
+	err = cl.Create(context.TODO(), pod)
+	Expect(err).NotTo(HaveOccurred())
+	return pod
+}
+
+func deletePod(pod runtime.Object) {
+	cl, err := client.New(cfg, client.Options{})
+	Expect(err).NotTo(HaveOccurred())
+	err = cl.Delete(context.TODO(), pod)
+	Expect(err).NotTo(HaveOccurred())
+}
+
 var _ = Describe("Informer Cache", func() {
-	var stop chan struct{}
 
 	BeforeEach(func() {
 		stop = make(chan struct{})
 		Expect(cfg).NotTo(BeNil())
+
+		By("creating three pods")
+		knownPod1 = createPod("test-pod-1", "test-namespace-1", kcorev1.RestartPolicyNever)
+		knownPod2 = createPod("test-pod-2", "test-namespace-2", kcorev1.RestartPolicyAlways)
+		knownPod3 = createPod("test-pod-3", "test-namespace-2", kcorev1.RestartPolicyOnFailure)
+
+		By("creating the informer cache")
+		var err error
+		informerCache, err = cache.New(cfg, cache.Options{})
+		Expect(err).NotTo(HaveOccurred())
+		By("running the cache and waiting for it to sync")
+		go func() {
+			defer GinkgoRecover()
+			Expect(informerCache.Start(stop)).ToNot(HaveOccurred())
+		}()
+		Expect(informerCache.WaitForCacheSync(stop)).NotTo(BeFalse())
 	})
+
 	AfterEach(func() {
+		By("cleaning up created pods")
+		deletePod(knownPod1)
+		deletePod(knownPod2)
+		deletePod(knownPod3)
+
 		close(stop)
 	})
 
-	Describe("as a Reader", func() {
-		It("should be able to list objects that haven't been watched previously", func() {
-			By("Creating the cache")
-			reader, err := cache.New(cfg, cache.Options{})
-			Expect(err).NotTo(HaveOccurred())
-
-			By("running the cache and waiting for it to sync")
-			go func() {
-				defer GinkgoRecover()
-				Expect(reader.Start(stop)).ToNot(HaveOccurred())
-			}()
-			Expect(reader.WaitForCacheSync(stop)).NotTo(BeFalse())
-
-			By("Listing all services in the cluster")
+	Describe("Cache as a Reader", func() {
```

Comment:
personal preference (totally fine to ignore): it's fine to have this just be `as a Reader`, since it's already in the context of `Informer Cache`